drivers/video/video_stm32_dcmi.c

```diff
@@ -109,6 +153,44 @@ void HAL_DMA_ErrorCallback(DMA_HandleTypeDef *hdma)
 	LOG_WRN("%s", __func__);
 }
 
+#if defined(CONFIG_SOC_SERIES_STM32U5X)
+static int stm32_dma_list_init(DMA_HandleTypeDef *hdma)
+{
+	static DMA_NodeTypeDef Node;
+	static DMA_QListTypeDef Queue;
+
+	HAL_StatusTypeDef ret = HAL_OK;
+	DMA_NodeConfTypeDef pNodeConfig;
+
+	/* Set node configuration ################################################*/
+	pNodeConfig.NodeType = DMA_GPDMA_LINEAR_NODE;
+	pNodeConfig.Init.Request = GPDMA1_REQUEST_DCMI_PSSI;
+	pNodeConfig.Init.BlkHWRequest = DMA_BREQ_SINGLE_BURST;
+	pNodeConfig.Init.Direction = DMA_PERIPH_TO_MEMORY;
+	pNodeConfig.Init.SrcInc = DMA_SINC_FIXED;
+	pNodeConfig.Init.DestInc = DMA_DINC_INCREMENTED;
+	pNodeConfig.Init.SrcDataWidth = DMA_SRC_DATAWIDTH_WORD;
+	pNodeConfig.Init.DestDataWidth = DMA_DEST_DATAWIDTH_WORD;
+	pNodeConfig.Init.SrcBurstLength = 1;
+	pNodeConfig.Init.DestBurstLength = 1;
+	pNodeConfig.Init.TransferAllocatedPort = DMA_SRC_ALLOCATED_PORT0 | DMA_DEST_ALLOCATED_PORT0;
+	pNodeConfig.Init.TransferEventMode = DMA_TCEM_BLOCK_TRANSFER;
+	pNodeConfig.TriggerConfig.TriggerPolarity = DMA_TRIG_POLARITY_MASKED;
+	pNodeConfig.DataHandlingConfig.DataExchange = DMA_EXCHANGE_NONE;
+	pNodeConfig.DataHandlingConfig.DataAlignment = DMA_DATA_RIGHTALIGN_ZEROPADDED;
+	pNodeConfig.SrcAddress = 0;
+	pNodeConfig.DstAddress = 0;
+	pNodeConfig.DataSize = 0;
+
+	ret |= HAL_DMAEx_List_BuildNode(&pNodeConfig, &Node);
+	ret |= HAL_DMAEx_List_InsertNode_Tail(&Queue, &Node);
+	ret |= HAL_DMAEx_List_SetCircularMode(&Queue);
+	ret |= HAL_DMAEx_List_LinkQ(hdma, &Queue);
+
+	return ret;
```

Comment:
using `ret |= ` is very compact, but if there are different error messages coming, then `return ret;` would become inaccurate, and then code that does `do { ret = try_again(); } while (ret == -EAGAIN);` can behave differently.